lib/src/cancelable_operation.dart

```diff
@@ -182,7 +241,7 @@ class CancelableOperation<T> {
             : (Object error, StackTrace stack) {
                 if (completer.isCanceled) return;
                 try {
-                  completer.complete(onError(error, stack));
+                  onError(error, stack, completer);
                 } catch (error2, stack2) {
                   completer.completeError(error2, stack2);
```

Comment:
So, here, if we want to work exactly like `Future.then`, we can do:
```dart
 completer.completeError(error2, identical(error, error2) ? stack : stack2);
```
(It's a silly thing, and these days you can just do `Error.throwWithStackTrace(e, s)` to explicitly ask for the same stack trace, so the functionality isn't strictly necessary.)

Reply:
Done.